app/routing/Versions.scala

```diff
@@ -35,11 +35,17 @@ object Version {
 
   object VersionReads extends Reads[Version] {
 
+    /** @param version
+     * expecting a JsString e.g. "1.0"
+     */
     override def reads(version: JsValue): JsResult[Version] =
-      version.validate[String].flatMap {
-        case Version1.name => JsSuccess(Version1)
-        case _             => JsError("Unrecognised version")
-      }
+      version
+        .validate[String]
+        .flatMap(name =>
+          Versions.getFrom(name) match {
+            case Left(_) => JsError("Version not recognised")
```

Comment:
as this is a code cleanup, if you reformat, should the JsError and JsSuccess be lined up?